src/application/application.ts

```diff
@@ -25,13 +27,35 @@ class Application {
             this.api.ExecuteTask(req, handleAPIResponse(resolve, reject))
         })
     }
+
+    executeTasKAndWaitFirstResult(req: ExecuteTaskRequest): Promise<ResultData> {
+        return new Promise<ResultData>((resolve, reject) => {
+            const id = uuidv4()
+            const stream = this.listenResult({ serviceID: req.serviceID, tagFilters: [id] })
+                .on('error', (err) => { reject(err) })
+                .on('metadata', (metadata) => {
+                    const err = checkStreamReady(metadata)
+                    if (err) return
+                    if (req.executionTags) {
+                        req.executionTags.push(id)
+                    } else {
+                        req.executionTags = [id]
+                    }
+                    this.executeTask(req).catch((err) => { stream.destroy(err) })
+                })
+                .on('data', (result) => { resolve(result) })
+        })
+    }
 }
 
 declare interface Stream<T> {
```

Comment:
Maybe the Stream interface could be put in a common type definition folder to be used both by application and service?

Reply:
https://github.com/mesg-foundation/mesg-js/issues/62